### PR TITLE
feat: add `privatek8s` AKS cluster

### PR DESCRIPTION
**Pull request by lemeurherve**
Ref: https://github.com/jenkins-infra/helpdesk/issues/2844

<details><summary>previously:</summary>

Diff between ext_jenkins-infra.yaml & ext_jenkins-infra_privatek8s.yaml temporary introduced until the switch from temp_privatek8s and privatek8s cluster for infra.ci.jenkins.io:

```diff
--- ext_jenkins-infra.yaml
+++ ext_jenkins-infra_privatek8s.yaml
@@ -1,3 +1,4 @@
+## TODO: rename this file to ext_jenkins-infra.yaml and get rid of the old one when the switch from temp-privatek8s to privatek8s is done
 serviceAccount:
   create: true
   name: jenkins-controller
@@ -40,10 +41,6 @@
     readinessProbe:
       initialDelaySeconds: 120
   testEnabled: false
-  podSecurityContextOverride:
-    runAsNonRoot: true
-    runAsUser: 1000
-    supplementalGroups: [1000]
   overwritePlugins: true
   serviceType: "ClusterIP"
   javaOpts: >-
@@ -88,10 +85,10 @@
                     username: "MY-USERNAME"
                 - file:
                     fileName: "kubeconfig"
-                    id: "kubeconfig-controller-temp-privatek8s"
-                    description: "Kubeconfig file for temp-privatek8s"
+                    id: "kubeconfig-controller-privatek8s"
+                    description: "Kubeconfig file for privatek8s"
                     scope: SYSTEM
-                    secretBytes: "${base64:${KUBECONFIG_TEMP_PRIVATEK8S}}"
+                    secretBytes: "${base64:${KUBECONFIG_PRIVATEK8S}}"
                 - basicSSHUserPrivateKey:
                     scope: SYSTEM
                     id: "ec2-agents-privkey"
@@ -120,7 +117,7 @@
                 namespace: "jenkins-agents"
                 podRetention: "Never"
                 serverUrl: ${AGENTS_KUBERNETES_CLUSTER_URL}
-                credentialsId: "kubeconfig-controller-temp-privatek8s"
+                credentialsId: "kubeconfig-controller-privatek8s"
                 serverCertificate: |
                   ${AGENTS_KUBERNETES_CLUSTER_CERT}
                 podLabels:

```

</details>

**About the removal of `podSecurityContextOverride` block:**
Adding this override the default values, which are fine as they are. Crucially, this overriding block is missing `fsGroup: 1000` default Jenkins Helm chart value, resulting in permission error in the initContainer when trying to deploy with it.

### CODE REVIEW

File: config/ext_private-nginx-ingress__common.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      emptyDir: {}
+    - name: nginx-logs
+      emptyDir: {}
+controller:
+  config:
+    log-format-upstream: '[$proxy_add_x_forwarded_for] - $remote_user [$time_local] "$request" $status $body_bytes_sent "$http_referer" "$http_user_agent" $request_length $request_time [$proxy_upstream_name] $upstream_addr $upstream_response_length $upstream_response_time $upstream_status $req_id'
+    # In order to use geoIP from the ingress controller,
+    # we need to provide a maxmind license key.
+    # I doubt we need it at the moment, hence this comment.
+    # https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#use-geoip2
```
Comment (lemeurherve) [inline config/ext_private-nginx-ingress__common.yaml:30]:
```suggestion
```
Not sure we'll do something about this (?)

File: config/ext_public-nginx-ingress__common.yaml [hunk 2]
```diff
@@ -22,9 +22,32 @@ defaultBackend:
...
       emptyDir: {}
 controller:
-  publishService:
-    enabled: true
+  config:
+    log-format-upstream: '[$proxy_add_x_forwarded_for] - $remote_user [$time_local] "$request" $status $body_bytes_sent "$http_referer" "$http_user_agent" $request_length $request_time [$proxy_upstream_name] $upstream_addr $upstream_response_length $upstream_response_time $upstream_status $req_id'
+    # In order to use geoIP from the ingress controller,
+    # we need to provide a maxmind license key.
+    # I doubt we need it at the moment, hence this comment.
+    # https://kubernetes.github.io/ingress-nginx/user-guide/nginx-configuration/configmap/#use-geoip2
```
Comment (lemeurherve) [inline config/ext_public-nginx-ingress__common.yaml:30]:
```suggestion
```
Same as above

File: config/ext_public-nginx-ingress__common.yaml [hunk 3]
```diff
@@ -22,9 +22,32 @@ defaultBackend:
...
+    use-geoip2: "true"
+    hsts: "true"
+    hsts-preload: "true"
+    hsts-include-subdomains: "true"
+    # Should be 2 years as per https://hstspreload.org/
+    # 24 hours, as a first gradual HSTS deployment - see INFRA-2998
+    hsts-max-age: "86400"
```
Comment (lemeurherve) [inline config/ext_public-nginx-ingress__common.yaml:37]:
```suggestion
    hsts-max-age: "2592000"
```
> **Strict-Transport-Security** The "max-age" directive is too small. The minimum recommended value is 2592000 (30 days).

**Review by dduportal — CHANGES_REQUESTED** [review]
As discussed together, we shall break down this PR

**Comment by lemeurherve** [comment]
PR broke down:
- #3233
- #3235
- #3247 (in draft while setting up a dual stack network)
- infra part in a following PR

**Comment by lemeurherve** [comment]
I was wondering if I could easily reuse this PR to include the remaining changes, nope ^^

image

**Pull request closed without merging**